### PR TITLE
Fix README typos, grammar

### PR DESCRIPTION
**Pull request by hunghvu**
~~Currently, `useFirebaseAdminDefaultCredential` is not in the `InitConfig` interface, so this causes `Object literal may only specify known properties` in TypeScript projects. This PR adds that property to the `InitConfig` interface.~~

- Address miscellaneous typos, grammar in either the documents or code samples.

---

Side question: Running a test on my environment results in carriage return error in Prettier (`error  Delete ␍  prettier/prettier`).  I temporarily added `    'prettier/prettier': ['error', { endOfLine: 'auto' }],`  in `.eslintrc.js` to workaround. To avoid modifying testing configuration, I did commit that, but should it be included in the PR?

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -67,6 +67,7 @@ interface InitConfig {
   tokenChangedHandler?: (user: AuthUser) => void
   onLoginRequestError?: (error: unknown) => void
   onLogoutRequestError?: (error: unknown) => void
+  useFirebaseAdminDefaultCredential?: boolean
```
Comment (kmjennison) [inline index.d.ts:70]: Please break this out into its own PR.

File: README.md [hunk 2]
```diff
@@ -33,7 +33,7 @@ We treat the Firebase JS SDK as the source of truth for auth status. When the us
 
 Depending on your app's needs, other approaches might work better for you.
 
-**If your app only uses static pages** or doesn't need the Firebase user for SSR, use the Firebase JS SDK directly to load the user on the client side.
+**If your app only uses static pages** or doesn't need the Firebase user for SSR, use the Firebase JS SDK directly to load the user on the client-side.
```
Comment (kmjennison) [inline README.md:36]: Splitting hairs here, but we aim to use "client side" for the noun and "client-side" for the adjective. Please keep nouns as-is.
https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/c/client-side

Comment (kmjennison) [reply README.md:36]: Same for "server-side" and "server side".

File: README.md [hunk 3]
```diff
@@ -248,8 +248,8 @@ It accepts the following options:
...
-| `whenUnauthedAfterInit`    | The action to take if the user is _not_ authenticated and the Firebase client JS SDK has already initialized. One of: `AuthAction.RENDER`, `AuthAction.REDIRECT_TO_LOGIN`.                                                                      | `AuthAction.RENDER`      |
+| `whenUnauthedBeforeInit`   | The action to take if the user is _not_ authenticated but the Firebase client JS SDK has not yet been initialized. One of: `AuthAction.RENDER`, `AuthAction.REDIRECT_TO_LOGIN`, `AuthAction.SHOW_LOADER`.                                            | `AuthAction.RENDER`      |
+| `whenUnauthedAfterInit`    | The action to take if the user is _not_ authenticated and the Firebase client JS SDK has already been initialized. One of: `AuthAction.RENDER`, `AuthAction.REDIRECT_TO_LOGIN`.                                                                      | `AuthAction.RENDER`      |
```
Comment (kmjennison) [inline README.md:252]: There's a distinction between the developer having initialized it vs. Firebase completing its initialization. We're trying to convey the latter, which I think is clearer without "been". Please revert.

File: README.md [hunk 4]
```diff
@@ -498,25 +498,25 @@ See the [default handler](https://github.com/gladly-team/next-firebase-auth/blob
...
+The `firebaseAdminInitConfig.credential.privateKey` cannot be defined on the client-side and should live in a secret environment variable.
 
 > ℹ️ Using Vercel? See [adding a private key to Vercel](#adding-a-private-key-to-Vercel) for guidance.
 
 #### useFirebaseAdminDefaultCredential
 
 `Boolean`
 
-When true, `firebase-admin` will use the Google Cloud application default credentials during [`initializeApp`](https://firebase.google.com/docs/admin/setup#initialize-sdk).
+When true, `firebase-admin` will implicitly find your hosting environment service account during `initializeApp`. This is applicable for both [Firebase](https://firebase.google.com/docs/admin/setup#initialize-sdk), and [Google Cloud Platform](https://cloud.google.com/docs/authentication/production), and more recommended than adding service account key to the code via file path or direct value.
```
Comment (kmjennison) [inline README.md:517]: Please change: "more recommended than adding" --> "recommended over adding"

File: README.md [hunk 5]
```diff
@@ -626,15 +626,15 @@ An async function that resolves to a valid Firebase ID token string, or null if
 
 **clientInitialized** - `Boolean`
 
-Whether the Firebase JS SDK has initialized. If `true`, we are no longer using any user info from server-side props.
+Whether the Firebase JS SDK has been initialized. If `true`, we are no longer using any user info from server-side props.
```
Comment (kmjennison) [inline README.md:629]: Same as above about "been initialized".

File: README.md [hunk 6]
```diff
@@ -908,7 +908,7 @@ export default getMockAuthUser
 
 Now, you can use and customize the mock behavior in your tests.
 
-If you're modifying higher-order functions, component being tested needs to be `require`d inside a `beforeEach` function or inside each test case. This is because mocking `next-firebase-auth` has to happen _before_ your component is imported, because the call to the `next-firebase-auth` function is part of the default export of your component (e.g., `export default withAuthUser()(MyComponent)`).
+If you're modifying higher-order functions, components being tested will be `required` inside a `beforeEach` function or each test case. This is because mocking `next-firebase-auth` has to happen _before_ your component is imported, because the call to the `next-firebase-auth` function is part of the default export of your component (e.g., `export default withAuthUser()(MyComponent)`).
```
Comment (kmjennison) [inline README.md:911]: Change: "will be" --> "need to be". This makes it clear the developer is responsible.

**Review by kmjennison — CHANGES_REQUESTED** [review]
Hi, thanks for the fixes and edits! Please:
* break out the type addition into its own PR
* see feedback on edits to the README

File: README.md (hunk 3 above)
Comment (hunghvu) [reply README.md:252]: Interesting, my initial interpretation was about the process of Firebase **being** initialized by `next-firebase-auth`, hence the word `been`. I will revert this.

**Comment by kmjennison** [comment]
Interesting that you're seeing a Prettier error locally when linting passes here. Maybe you're running a different version of Prettier from a global install? Regardless, doesn't look like it's an issue for this PR. Thanks for the improvements!

**Review by kmjennison — APPROVED** [review]
LGTM
